### DISCUSSION TITLE
is there a token limit to the document size you ingest?

### DISCUSSION THREAD

**Discussion by Deph1** [imartinez/privateGPT#400] [General]
As the title suggests, I couldn't see any information that dictates a token size limit. 

Thank you!

**Comment by johnbrisbin** [comment]
There are no limits beyond memory and disk space because any file is reduced to a text stream which is then chopped up into manageable sized pieces of 500 bytes or so. When queries are made the appropriate sections are located and passed to the LLM for it to draw conclusions from. The chunks that are passed to the LLM as part of the prompt must be, in sum, less than the context size minus some space for the answer in the output.

**Comment by KeulzZz** [comment]
I saw a considerable difference in ingest between PDFs and those PDFs converted to txt files, and a 10MB pdf always failling the ingest.
